src/recipe.c

```diff
@@ -445,7 +446,15 @@ recipe_parse (xmlDoc *doc, SoupURI *recipe_uri, GError **error, gchar **cfg_file
         g_free (tmp_str);
     }
     result->recipe_uri = recipe_uri;
-    result->base_path = TASK_LOCATION;
+    // Check if restraint-rhts is installed
+    rhts_vers = get_package_version("restraint-rhts", &tmp_error);
```

Comment:
This is actually not good solution for what we are trying to sell.
`Restraint` shouldn't care at all about `Restraint-RHTS`.
I will install Restraint and do my thing. I will install Restraint-RHTS and I'm expecting I will get modified behavior of `Restraint`. But In my opinion, Restraint should provide interface to change behavior. Rather than having logic for wrappers. Let's say I would like to wrap Restraint with something else (not RHTS). You will miss the interface again, therefore you can't change location.

There was a good idea before. Make `location` configurable. In that case we can set default in config. Replace it in RHTS and for tests use something completely different

Reply:
Even with Beaker from source code we have to change all locations.
Because devel version is not installed in same particular fs structures.

Reply:
I agree that this should be a config setting and not magically defined based on a package being installed or not.  restraint-rhts package could override the default config.

Reply:
ACK. I will replace this pull with another pull request since the changes are quite different.

Reply:
What do you have in mind with regard to config?  Recipe/task Param or via restraintd argc/v?  I'm thinking the former so the install path could end up varying per task?????

Carol

Reply:
I would say /var/lib/restraint/config.conf but @StykMartin might have another idea?

Reply:
Spoke with Martin and he wants a config file specifically for thiis purpose.  Also discussed this some with Daniel. My plan is to update specification file so the install directory is added to /var/lib/restraint/install_dir.  When installing rhts package, install dir will be /mnt/tests.  When installing non-rhts package, install dir will be /var/lib/restraint/tests.  With respect to installing by source files, it is the responsibility of the installer to create this file.  The 'restraintd' package will quit if the 'install_dir' file does not exist and an error message will be generated.